src/proof/alethe/alethe_post_processor.cpp

```diff
@@ -1488,6 +1488,102 @@ bool AletheProofPostprocessCallback::update(Node res,
   }
 }
 
+// The last step of the proof was:
+//
+// Children:  (P1:C1) ... (Pn:Cn)
+// Arguments: (AletheRule::VRULE,false,(cl false))
+// ---------------------
+// Conclusion: (false)
+//
+// In Alethe:
+//
+//  P1 ... Pn
+// ------------------- VRULE   ---------------------- FALSE
+//  (VP1:(cl false))*           (VP2:(cl (not true)))
+// -------------------------------------------------- RESOLUTION
+//                       (cl)**
+//
+// *  the corresponding proof node is ((false))
+// ** the corresponding proof node is (false)
+bool AletheProofPostprocessCallback::finalStep(
+    Node res,
+    PfRule id,
+    const std::vector<Node>& children,
+    const std::vector<Node>& args,
+    CDProof* cdp)
+{
+  NodeManager* nm = NodeManager::currentNM();
+
+  if (
+      // If the last proof rule was not translated yet
+      (id == PfRule::ALETHE_RULE) &&
+      // This case can only occur if the last step is an assumption
+      ((args[2].end() - args[2].begin()) > 1) &&
+      // If the proof node has result (false) additional steps have to be added.
+      (args[2][1].toString() != nm->mkConst(false).toString()))
```

Comment:
Since you're using the node for `false` more than once, create it in the beginning of the method and use that.

Also when you caompare nodes you don't need to convert them to strings.